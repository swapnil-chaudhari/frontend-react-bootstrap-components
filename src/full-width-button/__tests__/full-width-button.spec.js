import React from 'react';
import renderShallow from 'render-shallow';
import noop from 'src/utils/noop';
import FullWidthButton from '../full-width-button';

describe('<FullWidthButton/>', () => {
    describe('when it renders', () => {
        const props = {
            handleClick: noop,
            buttonContent: 'Button',
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<FullWidthButton { ...props } />).output;
        });

        test('should render add-button div', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it renders with disabled true but without disabled layout', () => {
        const props = {
            handleClick: noop,
            buttonContent: 'Button',
            disabled: true
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<FullWidthButton { ...props } />).output;
        });

        test('should render add-button Button with disabled prop', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it renders with disabled true and with disabled layout', () => {
        const props = {
            handleClick: noop,
            buttonContent: 'Button',
            disabled: true,
            disabledLayout: <span>Some Custom Disabled Layout</span>
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<FullWidthButton { ...props } />).output;
        });

        test('should render add-button div without Button', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it renders with note', () => {
        const props = {
            handleClick: noop,
            buttonContent: 'Button',
            note: 'Some note'
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<FullWidthButton { ...props } />).output;
        });

        test('should render add-note div', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
