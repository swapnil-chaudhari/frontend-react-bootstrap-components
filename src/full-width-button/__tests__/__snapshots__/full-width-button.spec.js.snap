// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullWidthButton/> when it renders should render add-button div 1`] = `
<div
  className="FullWidthButton"
>
  <div
    className="FullWidthButton-button-container"
  >
    <Button
      className="FullWidthButton-button"
      disabled={false}
      onClick={[Function]}
    >
      Button
    </Button>
  </div>
</div>
`;

exports[`<FullWidthButton/> when it renders with disabled true and with disabled layout should render add-button div without Button 1`] = `
<div
  className="FullWidthButton"
>
  <div
    className="FullWidthButton-button-container"
  >
    <div
      className="FullWidthButton-disabled-layout"
    >
      <span>
        Some Custom Disabled Layout
      </span>
    </div>
  </div>
</div>
`;

exports[`<FullWidthButton/> when it renders with disabled true but without disabled layout should render add-button Button with disabled prop 1`] = `
<div
  className="FullWidthButton"
>
  <div
    className="FullWidthButton-button-container"
  >
    <Button
      className="FullWidthButton-button"
      disabled={true}
      onClick={[Function]}
    >
      Button
    </Button>
  </div>
</div>
`;

exports[`<FullWidthButton/> when it renders with note should render add-note div 1`] = `
<div
  className="FullWidthButton"
>
  <div
    className="FullWidthButton-button-container"
  >
    <Button
      className="FullWidthButton-button"
      disabled={false}
      onClick={[Function]}
    >
      Button
    </Button>
  </div>
  <div
    className="FullWidthButton-add-note"
  >
    Some note
  </div>
</div>
`;
